pkg/reconciler/kubernetes/adapter.go

```diff
@@ -91,93 +125,352 @@ func (g *kubeClientAdapter) Kubeconfig() string {
 }
 
 func (g *kubeClientAdapter) PatchUsingStrategy(kind, name, namespace string, p []byte, strategy types.PatchType) error {
-	_, _, err := g.kubeClient.PatchUsingStrategy(kind, name, namespace, p, strategy)
-	return err
+	metadata := &Metadata{}
+	gvk, err := g.mapper.KindFor(schema.GroupVersionResource{Resource: kind})
+	if err != nil {
+		return err
+	}
+
+	restMapping, err := g.mapper.RESTMapping(gvk.GroupKind(), gvk.Version)
+	if err != nil {
+		return err
+	}
+
+	restClient, err := newRestClient(*g.restConfig, gvk.GroupVersion())
+	if err != nil {
+		return err
+	}
+
+	helper := resource.NewHelper(restClient, restMapping)
+
+	var u *unstructured.Unstructured
+
+	if helper.NamespaceScoped {
+		u, err = g.dynamicClient.
+			Resource(restMapping.Resource).
+			Namespace(namespace).
+			Patch(context.TODO(), name, strategy, p, metav1.PatchOptions{})
+	} else {
+		u, err = g.dynamicClient.
+			Resource(restMapping.Resource).
+			Patch(context.TODO(), name, strategy, p, metav1.PatchOptions{})
+	}
+
+	if err != nil {
+		return err
+	}
+
+	gvr := restMapping.Resource
+
+	metadata.Name = u.GetName()
+	metadata.Namespace = u.GetNamespace()
+	metadata.Group = gvr.Group
+	metadata.Resource = gvr.Resource
+	metadata.Version = gvr.Version
+	metadata.Kind = gvk.Kind
+	return nil
 }
 
-func (g *kubeClientAdapter) Deploy(ctx context.Context, manifest, namespace string, interceptors ...ResourceInterceptor) ([]*Resource, error) {
+func (g *kubeClientAdapter) DeployByCompareWithOriginal(ctx context.Context, manifestOriginal, manifestTarget, namespace string, interceptors ...ResourceInterceptor) ([]*Resource, error) {
 	if namespace == "" {
 		namespace = defaultNamespace
 	}
 
-	deployedResources, err := g.deployManifest(ctx, manifest, namespace, interceptors)
+	resourceInfoOriginal, err := g.helmClient.Build(bytes.NewBuffer([]byte(manifestOriginal)), false)
+	if err != nil {
+		g.logger.Errorf("Failed to process original manifest data for deploy: %s", err)
+		g.logger.Debugf("Manifest data: %s", manifestOriginal)
+		return nil, err
+	}
+
+	resourceInfoTarget, err := g.convertToResourceInfoWithInterceptors(manifestTarget, namespace, interceptors)
+	if err != nil {
+		g.logger.Errorf("Failed to process target manifest data for deploy: %s", err)
+		g.logger.Debugf("Manifest data: %s", manifestTarget)
+		return nil, err
+	}
+
+	deployedResources, err := g.deployResources(ctx, resourceInfoOriginal, resourceInfoTarget, namespace)
 
-	//delete namespace if no resources was deployed into it
 	if len(deployedResources) == 0 {
-		g.logger.Warnf("Namespace '%s' was required for deploying the manifest "+
+		g.logger.Warnf("Namespace '%s' was required for deploying the manifestTarget "+
 			"but no resources were finally deployed into it", namespace)
 	}
 
+	// TODO: consider if make sense to delete resources which in resourceInfoOriginal but not in resourceInfoTarget.
```

Comment:
**To reviewer:** 
This logic we need to discuss if necessary to have, but currently, there is no scenario to use this `DeployByCompareWithOriginal` method, should be fine.